torch/csrc/autograd/python_variable.cpp

```diff
@@ -2070,6 +2087,72 @@ static int THPVariable_subclass_traverse(
           }
         }
       }
+      // Symbolic sizes can contain Python objects, make sure to traverse them
+      // NB: we don't use the public APIs since those can trigger arbitrary
+      // Python code
+      if (auto* extra_meta =
+              tensor.unsafeGetTensorImpl()->maybe_get_extra_meta()) {
+        if (auto* sym_meta = extra_meta->symbolic_shape_meta_.get()) {
+          auto extract_symint =
+              [](const SymInt& s) -> PyObject* {
```

Comment:
These utilities don't already exists? :thinking: 
Or you want special properties here (like make sure there is no side effects)?

Reply:
This is a very rare thing to want to do from C++; normally you are happy to have the SymNodeImpl and you go through the virtual interface. It's only from Python GC related functions that you need to get at the PyObject.